internal/armada/server/event.go

```diff
@@ -49,13 +53,52 @@ func (s *EventServer) Report(ctx context.Context, message *api.EventMessage) (*t
 	if err := checkPermission(s.permissions, ctx, permissions.ExecuteJobs); err != nil {
 		return nil, status.Errorf(codes.PermissionDenied, "[Report] error: %s", err)
 	}
+	if event, ok := message.Events.(*api.EventMessage_Preempted); ok {
+		if err := s.preemptedEventHandler(event); err != nil {
+			return &types.Empty{}, err
+		}
+	}
 	return &types.Empty{}, s.eventStore.ReportEvents([]*api.EventMessage{message})
 }
 
+func (s *EventServer) preemptedEventHandler(event *api.EventMessage_Preempted) error {
+	if event.Preempted.JobId != "" {
+		result, err := s.jobRepository.GetJobsByIds([]string{event.Preempted.JobId})
```

Comment:
This might be slow because we'll end up doing a (potentially) large number of single redis lookups.  It's probably not worth optimizing right now, but worth bearing in mind.